### PR TITLE
Implement adding comments to list-issues

### PR DESCRIPTION
**Pull request by MeBrei**
When viewing an issues via 'list-issues', a comment can be entered at the top.

### CODE REVIEW

**Comment by jameshaydon** [comment]
Some comments, not related to the current diff:

`(load! "rad/monadic/issues.rad")`
Maybe we should make the imports of `issues.rad` a little leaner so that this CLI tool is a little more performant.

```
(insert
           :state :open
           (insert
              :comments []
              (insert
                :created-at (now!)
iss)))
```

One can use `<>`.

```
(def key (nth 0 v))
(def val (nth 1 v))
```
Use `match`.

```
(dict-from-seq (map mk-key (seq (list-issues))))
```
I think `map-keys` can be used here.

File: bin/rad-issues [hunk 1]
```diff
@@ -26,19 +26,38 @@
       [(string-append (show key) " - " (lookup :title val)) val]))
     (def iss (dict-from-seq (map mk-key (seq (list-issues)))))
     (def selected (fzf-select-by-key! iss))
+    (def comment-divider "---")
+    (def username
+      (fn [d]
+        (catch 'any (lookup :github-username d) (fn [x] "???"))))
```
Comment (jameshaydon) [inline bin/rad-issues]: I think we have `@def` for this.

File: bin/rad-issues [hunk 2]
```diff
@@ -26,19 +26,38 @@
...
+    (def issue-number (fn [d] (lookup :number d)))
+    (def body (fn [d] (lookup :body d)))
     (def pretty-comment (fn [c]
       (string-append
-        "* **" (lookup :github-username c) "** [" (lookup :created-at c) "]:\n\n"
-        (string/unlines (map (fn [xs] (string-append "   " xs)) (string/lines (lookup :body c)))))))
+        "* **" (username c) "** [" (lookup :created-at c) "]:\n\n"
+        (string/intercalate "\n\n" (map (fn [xs] (string-append "   " xs)) (string/lines (body c)))))))
```
Comment (jameshaydon) [inline bin/rad-issues]: Why the empty lines? I.e., why not use `string/unlines`?

File: bin/rad-issues [hunk 3]
```diff
@@ -26,19 +26,38 @@
...
+         "\n## Comments\n\n\n" (string/unlines (map pretty-comment (lookup :comments i)))
     ))))
-    (print! (edit-in-editor! (pretty-issue selected)))))
+    (def edited-issue (edit-in-editor! (pretty-issue selected)))
+    (def extract-comment
+      (fn [xs]
+        (string/unlines (take-while (fn [x] (not (eq? x comment-divider))) xs))))
+    (def send-comment
+      (fn [xs]
+        (def comment (extract-comment (string/lines edited-issue)))
```
Comment (jameshaydon) [inline bin/rad-issues]: I might move the `string/lines` into `extract-comment`.

File: bin/rad-issues (hunk 2 above)
Comment (MeBrei) [reply bin/rad-issues]: `string/unlines` adds `\n` between the lines, this was not detected as a linebreak by the markdown preview though.

File: bin/rad-issues (hunk 1 above)
Comment (jameshaydon) [reply bin/rad-issues]: Actually, it's only added in #324.

File: bin/rad-issues (hunk 2 above)
Comment (jameshaydon) [reply bin/rad-issues]: Ah I see.

**Pull request merged**